### DISCUSSION TITLE
HSR main power switch

### DISCUSSION THREAD

**Discussion by ricardocannizzaro** [HumanSupportRobot/QandA#15] [Q&A] [answered]
Hi,

I'd like to know more about the main power switch on the HSR. 

From the [start-up instructions](https://docs.hsr.io/hsrb_user_manual_en/howto/on_off.html), I understand that the main power switch needs to be on to power up the HSR, and it is recommended to be in the Off state when the HSR is not in use. I also found the [startup switch document](https://docs.hsr.io/hsrb_user_manual_en/_downloads/startup_switch_en.pdf), which mentions a charge circuit microcontroller.

What does the main power switch do exactly? Does it isolate the battery from the power relay board? Does it isolate the battery from the battery charging circuit? Thus, is it possible and/or recommended to charge the HSR with the main power switch in the Off state? I don't want to try this without knowing whether this is recommended/safe or not.

Many thanks,
Ricardo Cannizzaro
Oxford Robotics Institute / University of Oxford / RoboCup@Home Team ORIon

**Comment by YutaItozawa** [comment, marked as answer]
Dear Ricardo Cannizzaro

My name is Yuta Itozawa. I work as a electrical engineer of HSR.
Sorry for my late reply.

The battery is not isolate physically from the power relay board.  Because we use MOS-FET to switch circuit not relay.
The battery will not be charged if the rocker switch(main power switch) turned off.
The main power switch is connected to the enable pin of theDC-DC converter  on the power relay board.
So, if the main power switch is off, the power relay does not work. Charging circuit also does not work.

The charging system of HSR is the almost the same as laptop PC.
If you plug the ACDC charger in HSR, the power relay board is up and charger circuit start to charging. 
The battery will be charged whether the system is activated by pressing the main power switch or not, as long as the charger is connected.

So, my recommendation is to charge the robot whenever you use it, whether the robot is powered on or not.
If the robot will not be used for a while, it is a good idea to disconnect the battery to prevent leak current from draining the battery.

best regards
Yuta Itozawa

**Reply by ricardocannizzaro** [reply]
Hi Yuta Itozawa,

Thank you for your detailed reply.

> If you plug the ACDC charger in HSR, the power relay board is up and charger circuit start to charging.
The battery will be charged whether the system is activated by pressing the main power switch or not, as long as the charger is connected.

Just to be clear, this assumes that the main power switch is in the On state? When you refer to the system being activated, you mean whether the **power button** is in the On state or not? Otherwise this would contradict the earlier statement: 

> The` battery will not be charged if the rocker switch(main power switch) turned off.

Perhaps the [startup switch document](https://docs.hsr.io/hsrb_user_manual_en/_downloads/startup_switch_en.pdf) could be expanded to include these details on under which conditions the battery can be charged, and the wiring of the battery and the main power switch.

Many thanks,
Ricardo

**Reply by YutaItozawa** [reply]
Dear Ricardo Cannizzaro

Oh, sorry I missed about it.

> If you plug the ACDC charger in HSR, the power relay board is up and charger circuit start to charging.
The battery will be charged whether the system is activated by pressing _**the main power switch**_ or not, as long as the charger is connected.

The power button is correct. It's not the main power switch.
Sorry for the confusion.

Best regards,
Yuta

**Reply by ricardocannizzaro** [reply]
Thanks for the clarification.
Best,
Ricardo
